libraries/core/src/main/kotlin/com/bumble/appyx/core/composable/PermanentChild.kt

```diff
@@ -0,0 +1,56 @@
+package com.bumble.appyx.core.composable
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import com.bumble.appyx.core.navigation.model.permanent.PermanentNavModel
+import com.bumble.appyx.core.node.Node
+import com.bumble.appyx.core.node.ParentNode
+
+@Composable
+fun <NavTarget : Any> ParentNode<NavTarget>.PermanentChild(
+    permanentNavModel: PermanentNavModel<NavTarget>,
+    navTarget: NavTarget,
+    decorator: @Composable (child: ChildRenderer) -> Unit
+) {
+    val child = remember(navTarget) {
+        permanentNavModel
+            .elements
+            .value
+            .find { it.key.navTarget == navTarget }
+            ?.let { childOrCreate(it.key) }
+            ?: throw IllegalStateException(
```

Comment:
Will this break the code within `StatefulNode1.kt`?

Just so I understand, it's now mandatory to pass in a `NavTarget` to the `PermanentNavModel` constructor?

Perhaps we should update `PermanentNavModel` to throw an `IllegalStateException` if there are no `NavTarget`'s provided? Could we also consider removing the vararg then, as this will definitely lead to issues

Reply:
good catch

Reply:
I think we should just ignore if there is no child of the specific type. Otherwise we break functionality where you add a child later. Or maybe provide some optional renderer function.

> Just so I understand, it's now mandatory to pass in a NavTarget to the PermanentNavModel constructor?

Why it should be so? We still have `add`/`addUnique` operations to add nodes later.

Reply:
I feel like if you have a logic of adding a child later at some point it shouldn't be a part of `PermanentNavModel`

Reply:
perhaps leave it as is for now, and consider changing this behaviour in 2.0?

Reply:
But is not it the case Lachlan wants? Load data and show permanent child only then with the loaded data?

Reply:
I can remove this check it so it works with `addUnique`  operation

Reply:
> But is not it the case Lachlan wants? Load data and show (add) permanent child only then with the loaded data?

I don't think so. If he doesn't add a key to `PermanentNavModel` associated node is not created, and therefore doesn't start loading its content

Reply:
@CherryPerry my usecase is that I have a PermanentChild that is added immediately (via the constructor) and is only shown once the data is loaded (unfortunately in my usecase the ParentNode is responsible for when to show the PermanentChild based on Output of whether the state is loaded or not)

Without this change, the PermanentChild would not correctly start the business logic when added via the PermanentChild constructor (well actually it can, but it renders the wrong one at the moment)

Reply:
Anyway, we are breaking existing use case without providing alternative, when you want to add your permanent child later after some flag is triggered. Now you can via `if (myCheck) PermanentChild(NavKey.AddMe)`.

Reply:
Or we can observe `permanentNavModel` in composition to render permanent children

```
val isAddMeExists = permanentNavModel
    .elements.
    .mapState(scope, SharingStarted.WhileSubscribed()) { navElements ->
         navElements
             .find { it.key.navTarget == navTarget } ?: false
     }
     ..collectAsState()

if (isAddMeExists) { PermanentChild() } 

```

Reply:
Perhaps, can we added to an overloaded PermanentChild while default one fails